metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/converter/ReferenceSequenceConverter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.importer.converter;
+
+import org.springframework.core.convert.converter.Converter;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.ReferenceSequence;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.Taxonomy;
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.ReferenceSequenceRepository;
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.TaxonomyRepository;
+import uk.ac.ebi.ena.sra.xml.AssemblyType;
+
+import java.util.Arrays;
+
+public class ReferenceSequenceConverter implements Converter<AssemblyType, ReferenceSequence> {
+
+    private TaxonomyRepository taxonomyRepository;
+    private ReferenceSequenceRepository referenceSequenceRepository;
+
+    public ReferenceSequenceConverter(TaxonomyRepository taxonomyRepository,
+                                      ReferenceSequenceRepository referenceSequenceRepository) {
+        this.taxonomyRepository = taxonomyRepository;
+        this.referenceSequenceRepository = referenceSequenceRepository;
+    }
+
+    @Override
+    public ReferenceSequence convert(AssemblyType assemblyType) {
+        Taxonomy taxonomy = findExistingorSaveandReturnTaxonomy(assemblyType);
+        return convertSaveAndReturnNewReferenceSequence(assemblyType, taxonomy);
+    }
+
+    private Taxonomy findExistingorSaveandReturnTaxonomy(AssemblyType assemblyType) {
```

Comment:
Sorry I was wrong before can you please separate converter and persistence ,  like i have done in importStudy's dependentObjects.

Reply:
Fixed in https://github.com/EBIvariation/metadata-ws/pull/50/commits/6ff97f90c3c951061edceeeccc4855a94e78f12a. Now the three processes are separated:

* ReferenceSequence conversion
* Taxonomy extraction, lookup & persistence
* ReferenceSequence lookup & persistence